### PR TITLE
NextHop WIP review 1

### PR DESCRIPTION
**Pull request by a-canary**
Review Only. Do not merge.
I want to get some eyes on this. Looking for high level feedback ATM.
I having trouble with makefiles. Haven't fully compiled yet.

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci].

File: iocore/hostdb/HostDB.cc [hunk 1]
```diff
@@ -77,6 +77,46 @@ HostDBCache hostDB;
...
+//////////////////////////////////////////////////////////////////////////////////////
+
+HostDBInfo *
+HostDBInfo::alloc(int size)
+{
+  size += sizeof(HostDBInfo);
+  int iobuffer_index = iobuffer_size_to_index(size);
+  ink_release_assert(iobuffer_index >= 0);
+  void *ptr = ioBufAllocator[iobuffer_index].alloc_void();
+  memset(ptr, 0, size);
```
Comment (SolidWallOfCode) [inline iocore/hostdb/HostDB.cc]: It shouldn't be necessary to do this - the constructor shouldn't assume zero'd memory.

File: iocore/hostdb/HostDB.cc [hunk 2]
```diff
@@ -77,6 +77,46 @@ HostDBCache hostDB;
...
+{
+  if (size < sizeof(HostDBInfo)) {
+    return nullptr;
+  }
+  HostDBInfo *ret = HostDBInfo::alloc(size - sizeof(HostDBInfo));
+  int buf_index   = ret->iobuffer_index;
+  memcpy((void *)ret, buf, size);
+  // Reset the refcount back to 0, this is a bit ugly-- but I'm not sure we want to expose a method
+  // to mess with the refcount, since this is a fairly unique use case
+  ret                 = new (ret) HostDBInfo();
```
Comment (SolidWallOfCode) [inline iocore/hostdb/HostDB.cc]: I'd be more tempted to have a constructor that takes the serialized buffer and not do the `memcpy`. That lets you reset in a constructor, which is much more understandable.

File: iocore/nexthop/NextHopHealth.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include "NextHopHost.h"
+
+/// ----------------------------------
+/// Health Check plugin API
+/// TODO: move to another file
+
+/// Define the types of health checks, or reasons to be down.
+enum HealthCheck_t : uint8_t {
```
Comment (SolidWallOfCode) [inline iocore/nexthop/NextHopHealth.h]: `enum class HealthCheck` and drop the `HC_` prefix? In general, TS doesn't use `_t` for classes (in fact, I think [it is forbidden](https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html)).

**Comment by a-canary** [comment]
moving this review to my fork

**Pull request closed without merging**